From 9bc2912988a2e775252571b718fa5cf47cd59e3a Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 22 Nov 2021 20:16:50 +0200
Subject: [PATCH] [SW-56233] security: gaudi2: fix prot bits loop

The prot bit config function loop, sub_minor index
value needs to be incremented for each iteration.
Protection bit config routine is generic such that
addition of new blocks to initialize would be easy.

Change-Id: I83545ab02bd7526e406bf4bae273d79eb6373035
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 5dd97c9..7918a3f 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -432,6 +432,7 @@
 				do {
 					init_prot_bits_block(cfg_idx, major_idx, minor_idx,
 							     sub_minor_idx);
+					sub_minor_idx++;
 				} while (sub_minor_idx < block_addr.sub_minor);
 				minor_idx++;
 			} while (minor_idx < block_addr.minor);
